### PR TITLE
Catching errors and logging output

### PR DESCRIPTION
**Pull request by GreenGremlin**
I tested this by checking out the branch on the pi and running it. Thinking about leaving it checkout for tonight's sunset.

### CODE REVIEW

**Comment by GreenGremlin** [comment]
It didn't encounter an error, so the error handling hasn't really been tested :\

File: main.py [hunk 1]
```diff
@@ -38,6 +42,15 @@ def get_timelapse_filename(output_dir):
 
 
 def main():
+
+    if sys.stdout.isatty():
+        handler = logging.StreamHandler(sys.stdout)
+        logger.setLevel(logging.DEBUG)
+    else:
+        handler = logging.handlers.SysLogHandler(address='/dev/log')
```
Comment (GreenGremlin) [inline main.py:51]: This also hasn't been tested :\

File: goldenhour/timelapse/ffmpeg.py [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
+
+
 def compile_video(photos_dir, output_filename, photos_per_second=30):
-    print('compiling timelapse (photos per second: {photos_per_second})'.format(
+    logger.info('compiling timelapse (photos per second: {photos_per_second})'.format(
         photos_per_second=photos_per_second,
     ))
     # TODO ensure output_filename ends with .mp4
     photos_pattern = '{}/image%05d.png'.format(photos_dir)
-    # TODO check exit status
```
Comment (alanhussey) [inline goldenhour/timelapse/ffmpeg.py:10]: Thanks for fixing this!

File: main.py (hunk 1 above)
Comment (alanhussey) [reply main.py:51]: Has this been tested at this point?

File: main.py [hunk 3]
```diff
@@ -38,6 +43,15 @@ def get_timelapse_filename(output_dir):
...
+        logger.setLevel(logging.INFO)
+    logger.addHandler(handler)
```
Comment (alanhussey) [inline main.py:53]: I'm unfamiliar with `logging` – I assume this handler behavior propagates to the rest of the process?

File: goldenhour/timelapse/pi.py [hunk 4]
```diff
@@ -1,31 +1,44 @@
...
-        '-w', '1280',
-        '-h', '720',
-        #'--rotation', '180', # set this to correct for camera orientation
-        '--quality', '100',
-        # '--verbose',
-        # '--saturation', '50',
-        '-o', output_pattern,
-        '-e', 'png',
-    ])
+    file_list = os.listdir(output_dir)
```
Comment (alanhussey) [inline goldenhour/timelapse/pi.py]: It's a little confusing to have (and reuse) the variable `file_list` when we only want its length. Maybe inline it?

**Review by alanhussey — COMMENTED** [review]
Sorry it took me so long to get to this! It looks pretty good to me, just left a couple comments.

File: main.py (hunk 3 above)
Comment (GreenGremlin) [reply main.py:53]: Yes, it will persist for the the rest of the process.

File: goldenhour/timelapse/pi.py (hunk 4 above)
Comment (GreenGremlin) [reply goldenhour/timelapse/pi.py]: Done

File: main.py (hunk 1 above)
Comment (GreenGremlin) [reply main.py:51]: Yes, here's some output:

```
Mar 17 15:00:05 sunset-cam verifying twitter credentials
Mar 17 15:00:05 sunset-cam estimated video length: 30.0 seconds
Mar 17 15:00:05 sunset-cam Waiting 3 hours 17 minutes 26 seconds to start, 60 minutes before sunset
Mar 17 18:17:32 sunset-cam recording timelapse (duration: 7200, interval: 8, filename: output/timelapse_2019-03-17_000.mp4)
Mar 17 18:17:32 sunset-cam capturing one photo every 8 seconds for 7200 seconds
Mar 17 20:17:42 sunset-cam Captured 901 photos
Mar 17 20:17:42 sunset-cam compiling timelapse (photos per second: 30)
Mar 17 20:20:57 sunset-cam 🌅 sunset at 07:17PM#012#012☀️ clear#012🌡 58℉#012💨 winds about 6mph from the northwest#012🌁 3.13 miles of visibility
Mar 17 20:20:57 sunset-cam posting to twitter (status_text: 🌅 sunset at 07:17PM#012#012☀️ clear#012🌡 58℉#012💨 winds about 6mph from the northwest#012🌁 3.13 miles of visibility, media: output/timelapse_2019-03-17_000.mp4)
Mar 17 20:21:02 sunset-cam done!
```

**Comment by GreenGremlin** [comment]
This should be ready to go

**Pull request merged**